### DISCUSSION TITLE
Ubuntu installation doesn't have configuration files and plugins

### DISCUSSION THREAD

**Discussion by esseredigitale** [rabbitmq/rabbitmq-server#8968] [Other]
Hi,
I have installed on Ubuntu using instructions in https://www.rabbitmq.com/install-debian.html and the server seems to start correctly.
I have installed management plugin to, but at http://<ip>:15672 no response.
Searching for configuration file rabbitmq.conf and plugins folder I can't find them!
Where can I find them?
Best regards

**Comment by lukebakken** [comment]
> Searching for configuration file rabbitmq.conf and plugins folder I can't find them! Where can I find them?

You must create them yourself in `/etc/rabbitmq`

> I have installed management plugin to, but at http://:15672 no response.

* How did you enable the management plugin?
* What is the output of `curl -4vu guest:guest localhost:15672/api/overview` when run on the same server as RabbitMQ?

**Comment by esseredigitale** [comment]
I have enabled the management plugin and curl -4vu guest:guest localhost:15672/api/overview give me 
*   Trying 127.0.0.1:15672...
* Connected to localhost (127.0.0.1) port 15672 (#0)
* Server auth using Basic with user 'guest'
> GET /api/overview HTTP/1.1
> Host: localhost:15672
> Authorization: Basic Z3Vlc3Q6Z3Vlc3Q=
> User-Agent: curl/7.81.0
> Accept: */*
>
* Mark bundle as not supporting multiuse
< HTTP/1.1 200 OK

I'm trying to call the web interface from an IP on same subnet: how can I fix it?

**Reply by michaelklishin** [reply]
Networking, CORS configuration is covered in the plugin doc guide https://rabbitmq.com/management.html#configuration.

For remote access you must create a new user, tag it appropriately and grant it relevant permissions https://rabbitmq.com/access-control.html#default-state

**Comment by esseredigitale** [comment]
> Networking, CORS configuration is covered in the plugin doc guide https://rabbitmq.com/management.html#configuration.
> 

done.

> For remote access you must create a new user, tag it appropriately and grant it relevant permissions https://rabbitmq.com/access-control.html#default-state

done.

I have to create the configuration files.

**Reply by michaelklishin** [reply]
Yes, you have to create the configuration file(s) [such as `rabbitmq.conf`](https://rabbitmq.com/configure.html#verify-configuration-config-file-location).

**Comment by michaelklishin** [comment]
Once the plugin is enabled and you have created a configuration file, and you have created a new user with the appropriate tags and permissions (for remote access), you can then [inspect the effective configuration](https://rabbitmq.com/configure.html#verify-configuration-effective-configuration), see what `rabbitmq-diagnostics listeners` reports, and try contacting the HTTP API.

See [Troubleshooting Networking](https://rabbitmq.com/troubleshooting-networking.html) and [server logs](https://rabbitmq.com/logging.html) if you still cannot connect.

Please take it from here.

**Reply by lukebakken** [reply]
@esseredigitale I have edited @michaelklishin's comment. By mistake he left out an "and" that made it seem like RabbitMQ creates a configuration file for you. It does not - you must create the configuration files.

**Comment by esseredigitale** [comment]
> Once the plugin is enabled a config file is created, 

I can't find it

>and a new user with the appropriate tags and permissions was added (for remote access), you can [inspect the effective configuration]
done

>(https://rabbitmq.com/configure.html#verify-configuration-effective-configuration), see what `rabbitmq-diagnostics listeners` reports, and try contacting the HTTP API.
> 
> See [Troubleshooting Networking](https://rabbitmq.com/troubleshooting-networking.html) and [server logs](https://rabbitmq.com/logging.html) if you still cannot connect.
> 
> Please take it from here.

After the installation I don't find config ang plugin files.

**Reply by michaelklishin** [reply]
You must create the file yourself. Neither RabbitMQ nor the Debian package create an empty config file.

Another option is [copying and modifying this example file](https://github.com/rabbitmq/rabbitmq-server/blob/v3.12.x/deps/rabbit/docs/rabbitmq.conf.example) but it is extensive and will be difficult to quickly inspect.

Plugin files ship with RabbitMQ. You do not need to interact with with the files as [plugins are enabled using `rabbitmq-plugins`](https://rabbitmq.com/plugins.html).

**Comment by esseredigitale** [comment]
> You must create the file yourself. Neither RabbitMQ nor the Debian package create an empty config file.
> 

ok, clear

> Another option is [copying and modifying this example file](https://github.com/rabbitmq/rabbitmq-server/blob/v3.12.x/deps/rabbit/docs/rabbitmq.conf.example) but it is extensive and will be difficult to quickly inspect.
> 

I'll try it

> Plugin files ship with RabbitMQ. You do not need to interact with with the files as [plugins are enabled using `rabbitmq-plugins`](https://rabbitmq.com/plugins.html).

Sorry, but I can't find that directories. I find only /etc/rabbitmq/enabled-plugins but is empty

**Reply by michaelklishin** [reply]
The link above explains how the plugins are enabled. `rabbitmq-plugins list` then can be used to inspect their state (by default it assumes a local node is running).

You do not need to locate or do anything about the plugin directories.

**Comment by esseredigitale** [comment]
> The link above explains how the plugins are enabled. `rabbitmq-plugins list` then can be used to inspect their state (by default it assumes a local node is running).
> 
> You do not need to locate or do anything about the plugin directories.

ok, I made a new empty Ubuntu 22.04 VM and after an upgrade, I executed the CloudSmith install script:

``` bash
#!/bin/sh

sudo apt-get install curl gnupg apt-transport-https -y

curl -1sLf "https://keys.openpgp.org/vks/v1/by-fingerprint/0A9AF2115F4687BD29803A206B73A36E6026DFCA" | sudo gpg --dearmor | sudo tee /usr/share/keyrings/com.rabbitmq.team.gpg > /dev/null

curl -1sLf https://ppa1.novemberain.com/gpg.E495BB49CC4BBE5B.key | sudo gpg --dearmor | sudo tee /usr/share/keyrings/rabbitmq.E495BB49CC4BBE5B.gpg > /dev/null

curl -1sLf https://ppa1.novemberain.com/gpg.9F4587F226208342.key | sudo gpg --dearmor | sudo tee /usr/share/keyrings/rabbitmq.9F4587F226208342.gpg > /dev/null

sudo tee /etc/apt/sources.list.d/rabbitmq.list <<EOF

deb [signed-by=/usr/share/keyrings/rabbitmq.E495BB49CC4BBE5B.gpg] https://ppa1.novemberain.com/rabbitmq/rabbitmq-erlang/deb/ubuntu jammy main

deb-src [signed-by=/usr/share/keyrings/rabbitmq.E495BB49CC4BBE5B.gpg] https://ppa1.novemberain.com/rabbitmq/rabbitmq-erlang/deb/ubuntu jammy main

deb [signed-by=/usr/share/keyrings/rabbitmq.9F4587F226208342.gpg] https://ppa1.novemberain.com/rabbitmq/rabbitmq-server/deb/ubuntu jammy main

deb-src [signed-by=/usr/share/keyrings/rabbitmq.9F4587F226208342.gpg] https://ppa1.novemberain.com/rabbitmq/rabbitmq-server/deb/ubuntu jammy main

EOF

sudo apt-get update -y

sudo apt-get install -y erlang-base \
                        erlang-asn1 erlang-crypto erlang-eldap erlang-ftp erlang-inets \
                        erlang-mnesia erlang-os-mon erlang-parsetools erlang-public-key \
                        erlang-runtime-tools erlang-snmp erlang-ssl \
                        erlang-syntax-tools erlang-tftp erlang-tools erlang-xmerl

sudo apt-get install rabbitmq-server -y --fix-missing
```

and the broker starts correctly.
After this with `sudo rabbitmq-plugins enable rabbitmq_management` I should see management at http://<ip>:15762 but nothing.
What's the problem?

Playing `curl -4vu guest:guest localhost:15672/api/overview` gave me HTTP OK, but after tha creation of an user with administrative privileges and using a rabbitmq.conf with only these lines

``` ini
listeners.tcp.default = 5672
management.tcp.port = 15672
management.tcp.ip   = 10.1.1.2
```

the output of `curl -4vu admin:pwd localhost:15672/api/overview` is Connection refused.

I'm going nuts!
Thanks a lot

**Reply by lukebakken** [reply]
I don't have the time to create a Ubuntu 22.04 VM, but I quickly put together a Dockerfile based on 22.04 and the commands you posted in your comment:

https://github.com/lukebakken/rabbitmq-server-8968

The Management starts as expected when I run these two commands:

```
make build
make run
```


> privileges and using a rabbitmq.conf with only these lines
> listeners.tcp.default = 5672
> management.tcp.port = 15672
> management.tcp.ip   = 10.1.1.2
> the output of curl -4vu admin:pwd localhost:15672/api/overview is Connection refused.

You are limiting the management UI to listen on `10.1.1.2`, so using `localhost` _will not work_, as that IP address is `127.0.0.1`.

What is the output of `curl -4vu admin:pwd 10.1.1.2:15672/api/overview`?

**Reply by michaelklishin** [reply]
Please take a moment to learn about [how to format code blocks on GitHub](https://docs.github.com/en/get-started/writing-on-github/working-with-advanced-formatting/creating-and-highlighting-code-blocks)

**Reply by michaelklishin** [reply]
Our team cannot troubleshoot every user installation but there is a [guide on troubleshooting network connectivity](https://rabbitmq.com/troubleshooting-networking.html) and another one [for TLS-enabled connections](https://rabbitmq.com/troubleshooting-ssl.html).

You don't have to specify default ports in `rabbitmq.conf` and if you are new to RabbitMQ or
networking, do not bind any listeners to a specific IP address. Start with all defaults, then add configuration
as you need it, step by step.

**Comment by esseredigitale** [comment]
> Please take a moment to learn about [how to format code blocks on GitHub](https://docs.github.com/en/get-started/writing-on-github/working-with-advanced-formatting/creating-and-highlighting-code-blocks)

I'm sorry, sure I will do it

**Comment by esseredigitale** [comment]
Unfortunately I didn't have much time to implement the installation, so I switched to ActiveMQ and everything works fine.
Thank you all for the support.
Greetings
